src/zenml/artifacts/type_registry.py

```diff
@@ -44,7 +44,15 @@ def register_integration(
 
     def get_artifact_type(self, key: Type[Any]) -> List[Type["BaseArtifact"]]:
         """Method to extract the list of artifact types given the data type"""
-        return self._artifact_types[key]
+        if key in self._artifact_types:
```

Comment:
Is there a possibility that no artifact types are registered for some class? If yes I feel like this method should raise a `KeyError` instead of whatever happens when calling `pop()` on an empty list. (And the KeyError should probably be in the docstring)

Reply:
Given the current state of the implementation, when the execution reaches that point there is no way for that list to be empty. It is due to the fact that the entries of the `type_registry` mirror the entries of the `materializer_registry` in terms of keys and the `materializer_registry` gets called first. So if there are no compatible subclasses, `materializer_registry` will fail before the code reaches that point. However, the implementation may change in the future and I agree that it would make sense to cover the edge-cases as we build it up. I will create a short fix for it!

Reply:
Added the checks to the `type_registry` as well and updated the docstrings.

Reply:
Looks awesome now!